### PR TITLE
MAGN-9588 If Point (manipulator) node fails to execute and is selected, Dynamo crashes while trying to create manipulator

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

This fix is to prevent crash in DM when dropping a Point.ByGeometry node and LibG fails to load [http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-9588] (http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-9588). The cause of the crash was that in situations where LibG fails to load, and a DM enabled node (like Point.ByCoordinates or Curve.PointAtParameter) is dropped, other ProtoGeometry methods used in DM code would also be invoked. While the exceptions coming from the geometry nodes are caught, those from the methods used in code are not thus causing the crash.

The fix is to disable drawing of manipulators early on by checking for the node's `CachedValue` to be null.

### Declarations

Check these if you believe they are true

- [X] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.

### Reviewers

@sharadkjaiswal

### CODE REVIEW

File: src/DynamoManipulation/NodeManipulator.cs [hunk 1]
```diff
@@ -551,13 +555,22 @@ public bool IsEnabled()
...
             {
                 return false;
             }
 
             return active;
         }
+
+        public bool IsNodeValueNull()
+        {
+            if (Node.CachedValue == null || Node.CachedValue.IsNull)
```
Comment (sharadkjaiswal) [inline src/DynamoManipulation/NodeManipulator.cs]: you could have simply returned `return Node == null || Node.CachedValue == null || Node.CachedValue.IsNull;`

File: src/DynamoManipulation/NodeManipulator.cs [hunk 2]
```diff
@@ -47,7 +47,7 @@ public abstract class NodeManipulator : INodeManipulator
         private const double NewNodeOffsetY = 50;
         private bool active;
 
-        protected const double gizmoScale = 1.2;
```
Comment (sharadkjaiswal) [inline src/DynamoManipulation/NodeManipulator.cs]: I think const was OK, any specific reason to change it to readonly?

**Comment by sharadkjaiswal** [comment]
Thanks @aparajit-pratap for looking into it. Is there any other place we need to safeguard? After this fix do we see Gizmos drawn?

File: src/DynamoManipulation/NodeManipulator.cs (hunk 2 above)
Comment (aparajit-pratap) [inline src/DynamoManipulation/NodeManipulator.cs]: No specific reason, was trying some changes. Will revert.

**Comment by aparajit-pratap** [comment]
Thanks @sharadkjaiswal for reviewing this. I have verified that the Gizmos continue to work as before.

**Pull request merged**